### ISSUE TITLE
"Could not add font!" on Linux build

### ISSUE THREAD

**Issue by vestingz** [ahrm/sioyek#4]
Thanks for providing a Linux build. However, [the current version](https://github.com/ahrm/sioyek/tree/v0.31.3) seems to fail to render any PDF file I throw at it with following error:
```
could not add font!1config file change listener constructor called1starting index thread ...
figure indexing finished ...
Impossible to open Impossible to open Impossible to open Impossible to open texture: 4
```
qt5-3d 5.15.2
libmupdf 1.18.0
Arch Linux 5.12.15 x86_64

**Comment by ahrm** [comment]
Ok so I tried it on an archlinux virtual machine and it kind of worked except it couldn't compile the shaders because the OpenGL version on the machine was not sufficient (sioyek requires at least OpenGL 3.3). I suspect maybe your OpenGL version is old too? Would you pleae tell me the output of this command on your machine? :
`glxinfo | grep version`

**Comment by ahrm** [comment]
I did manage to fix it in virtual machine by following this:

https://bbs.archlinux.org/viewtopic.php?pid=1853919#p1853919

which seems to manually override the default OpenGL version. However, other users warned that:
"those settings are meant for developers and can lead to severe instabilities.
Remove those lines from /etc/environment."

and I am not an expert when it comes to archlinux so do it at your own risk.

**Comment by vestingz** [comment]
Thank you for your efforts! :)

Indeed, I am also running OpenGL 3.0. Seems like I should rather wait 
then for upstream updates.

This issue can be closed, I guess.

**Comment by ahrm** [comment]
> Thanks for providing a Linux build. However, [the current version](https://github.com/ahrm/sioyek/tree/v0.31.3) seems to fail to render any PDF file I throw at it with following error:
> 
> ```
> could not add font!1config file change listener constructor called1starting index thread ...
> figure indexing finished ...
> Impossible to open Impossible to open Impossible to open Impossible to open texture: 4
> ```
> 
> qt5-3d 5.15.2
> libmupdf 1.18.0
> Arch Linux 5.12.15 x86_64

I think I found the problem. I usually use the path of the executable as the parent path and put all the config files next to the executable. But I had the following code to change the parent directory in debug mode so I could experiment easily:
```
#ifdef NDEBUG
	install_app(exe_path.c_str());
#else
	std::filesystem::path source_file_path = __FILE__;
	parent_path = source_file_path.parent_path();
#endif
```

However, I think NDEBUG is a visual studio only definition, so when I build in linux, it uses the current directory as the parent directory. I think if you execute `AppRun` from the directory containing the `shaders` folder, it will work. I am working on a new linux build withut this problem.

**Comment by vestingz** [comment]
That sounds great, thanks for the update!

**Comment by ahrm** [comment]
> Thank you for your efforts! :) Indeed, I am also running OpenGL 3.0. Seems like I should rather wait then for upstream updates. This issue can be closed, I guess.

I will keep the issue open. Maybe I can reduce the minimum required OpenGL version to 3.0?

Although I don't think waiting for upstream updates is a valid solution, OpenGL 3.0 came out in 2008!

**Comment by ahrm** [comment]
I think this issue is fixed in 7025bdd71d5d623debc16e4c1752a7f073399344 .
